### PR TITLE
RANCHER-1360: Add Eureka support to folioHelm function

### PR DESCRIPTION
**Pull request by yaroslavishchenko**
Related ticket in Jira:
[RANCHER-1360: Add Eureka support to folioHelm function](https://folio-org.atlassian.net/browse/RANCHER-1360)

### CODE REVIEW

File: resources/helm/testing.yaml [hunk 1]
```diff
@@ -1766,3 +1766,93 @@ ui-bundle:
       memory: 512Mi
     requests:
       memory: 256Mi
+mod-consortia-keycloak:
+  extraEnvVars:
+  - name: SECRET_STORE_TYPE
```
Comment (eldiiar-duishenaliev) [inline resources/helm/testing.yaml:1771]: this is already a part of the eureka-common secret

File: resources/helm/testing.yaml [hunk 2]
```diff
@@ -1766,3 +1766,93 @@ ui-bundle:
...
+      existingSecret: mod-consortia-keycloak-systemuser
+  replicaCount: 1
+  resources:
+    limits:
+      memory: 1Gi
+    requests:
+      memory: 256Mi
+mod-login-keycloak:
+  extraEnvVars:
+    - name: SECRET_STORE_TYPE
```
Comment (eldiiar-duishenaliev) [inline resources/helm/testing.yaml:1794]: same is here

File: resources/helm/testing.yaml [hunk 3]
```diff
@@ -1766,3 +1766,93 @@ ui-bundle:
...
+  resources:
+    limits:
+      memory: 1Gi
+    requests:
+      memory: 256Mi
+mod-roles-keycloak:
+  extraEnvVars:
+    - name: KC_ADMIN_CLIENT_ID
+      value: "folio-backend-admin-client"
+    - name: SECRET_STORE_TYPE
```
Comment (eldiiar-duishenaliev) [inline resources/helm/testing.yaml:1816]: same is here

File: resources/helm/testing.yaml [hunk 4]
```diff
@@ -1766,3 +1766,93 @@ ui-bundle:
...
+  resources:
+    limits:
+      memory: 1Gi
+    requests:
+      memory: 256Mi
+mod-users-keycloak:
+  extraEnvVars:
+    - name: KC_ADMIN_CLIENT_ID
+      value: "folio-backend-admin-client"
+    - name: SECRET_STORE_TYPE
```
Comment (eldiiar-duishenaliev) [inline resources/helm/testing.yaml:1838]: same is here

File: vars/folioNamespaceCreate.groovy [hunk 5]
```diff
@@ -30,6 +30,7 @@ void call(CreateNamespaceParameters args) {
     tfConfig.addVar('pg_ldp_user_password', Constants.PG_LDP_DEFAULT_PASSWORD)
     tfConfig.addVar('github_team_ids', folioTools.getGitHubTeamsIds("${Constants.ENVS_MEMBERS_LIST[args.namespaceName]},${args.members}").collect { "\"${it}\"" })
```
Comment (eldiiar-duishenaliev) [inline vars/folioNamespaceCreate.groovy:31]: there should be conditional step for eureka related helm chart...

**Review by eldiiar-duishenaliev — CHANGES_REQUESTED** [review]

File: resources/helm/testing.yaml (hunk 1 above)
Comment (yaroslavishchenko) [reply resources/helm/testing.yaml]: Adjusted accordingly to your suggestion.
Thanks!

File: resources/helm/testing.yaml (hunk 2 above)
Comment (yaroslavishchenko) [reply resources/helm/testing.yaml]: Adjusted accordingly to your suggestion.
Thanks!

File: resources/helm/testing.yaml (hunk 3 above)
Comment (yaroslavishchenko) [reply resources/helm/testing.yaml]: Adjusted accordingly to your suggestion.
Thanks!

File: vars/folioNamespaceCreate.groovy (hunk 5 above)
Comment (yaroslavishchenko) [reply vars/folioNamespaceCreate.groovy:31]: @eldiiar-duishenaliev 
In repsonse to your request I've added conditional helm chart deployment.
Could you please take a look?

File: src/org/folio/Constants.groovy [hunk 6]
```diff
@@ -122,6 +122,10 @@ class Constants {
   static String FOLIO_HELM_V2_TEST_REPO_NAME = "folio-helm-v2-test"
   static String FOLIO_HELM_V2_TEST_REPO_URL = 'https://repository.folio.org/repository/folio-helm-v2-test/'
 
+  // Bitnami Helm Proxy Repository
+  static String BITNAMI_HELM_PROXY_REPO_NAME = "helm-bitnami-proxy"
```
Comment (eldiiar-duishenaliev) [inline src/org/folio/Constants.groovy:126]: We don't need these lines, because we do not use bitnami's helm chart for BE modules, already method exists: https://github.com/folio-org/pipelines-shared-library/blob/master/vars/folioHelm.groovy#L18

File: vars/folioHelm.groovy [hunk 7]
```diff
@@ -188,6 +188,11 @@ String generateModuleValues(RancherNamespace ns, String moduleName, String modul
     moduleConfig['extraEnvVars:'] += [name: 'LEGACY_TOKEN_TENANTS', value: '']
   }
 
+  // Switch on Eureka related configuration in Helm Charts for Modules
+  if (ns.enableEureka) {
+    moduleConfig << [ eureka : [enabled: true] ]
```
Comment (eldiiar-duishenaliev) [inline vars/folioHelm.groovy:193]: Modification is required, overwise sidecar will be inside all of the modules...
ONLY mod-* modules should have sidecar inside.

File: vars/folioHelmFlow.groovy [hunk 8]
```diff
@@ -4,6 +4,15 @@ import org.folio.models.RancherNamespace
 
 import java.time.LocalDateTime
 
+void deployEureka(String namespace) {
```
Comment (eldiiar-duishenaliev) [inline vars/folioHelmFlow.groovy:7]: Another portion of unrelated lines...

File: vars/folioNamespaceCreate.groovy [hunk 9]
```diff
@@ -35,6 +35,14 @@ void call(CreateNamespaceParameters args) {
       folioTerraformFlow.manageNamespace('apply', tfConfig)
     }
 
+    if (args.eureka) {
+      stage('[Helm] Deploy Eureka') {
+        folioHelm.withKubeConfig(namespace.getClusterName()) {
```
Comment (eldiiar-duishenaliev) [inline vars/folioNamespaceCreate.groovy:40]: Modification is required here. Will explain it today to you.

File: src/org/folio/Constants.groovy (hunk 6 above)
Comment (yaroslavishchenko) [reply src/org/folio/Constants.groovy:126]: Suggested constans are removed.

File: vars/folioHelmFlow.groovy (hunk 8 above)
Comment (yaroslavishchenko) [reply vars/folioHelmFlow.groovy:7]: Deployment steps are removed until it will be clear what to put here.

File: src/org/folio/Constants.groovy [hunk 10]
```diff
@@ -274,4 +274,11 @@ class Constants {
                                 'folio-module-sidecar',
                                 'folio-kong',
                                 'folio-keycloak']
+
+  static List EUREKA_MODULES_WITH_SIDECAR = [
+    'mod-scheduler',
```
Comment (eldiiar-duishenaliev) [inline src/org/folio/Constants.groovy:279]: Not required, please remove.

File: vars/folioHelm.groovy [hunk 11]
```diff
@@ -188,6 +188,11 @@ String generateModuleValues(RancherNamespace ns, String moduleName, String modul
     moduleConfig['extraEnvVars:'] += [name: 'LEGACY_TOKEN_TENANTS', value: '']
   }
 
+  // Switch on Eureka related configuration in Helm Charts for Modules
+  if (ns.enableEureka && moduleName in Constants.EUREKA_MODULES_WITH_SIDECAR) {
```
Comment (eldiiar-duishenaliev) [inline vars/folioHelm.groovy:192]: Will not work in our case

File: vars/folioHelm.groovy [hunk 12]
```diff
@@ -188,6 +188,11 @@ String generateModuleValues(RancherNamespace ns, String moduleName, String modul
     moduleConfig['extraEnvVars:'] += [name: 'LEGACY_TOKEN_TENANTS', value: '']
   }
 
+  // Switch on Eureka related configuration in Helm Charts for Modules
+  if (ns.enableEureka && moduleName.startsWith('mod-')) {
```
Comment (yaroslavishchenko) [inline vars/folioHelm.groovy:192]: @eldiiar-duishenaliev 

I have concern for recommended approach.
All common backend modules (not Eureka related) also start with "mod-" prefix.

Does it allign with our goals in this case?
